pkg/operator/ceph/csi/peermap/clientProfile.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peermap
+
+import (
+	"context"
+	"reflect"
+	"slices"
+
+	csiopv1a1 "github.com/ceph/ceph-csi-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ReconcileClientProfileMapping creates/updates a clientProfileMapping resource used by the ceph csi-operator
+func ReconcileClientProfileMapping(ctx context.Context, clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, pool *cephv1.CephBlockPool) error {
+	if pool.Spec.Mirroring.Peers == nil {
+		logger.Debugf("no peer secrets added in cephBlockPool %q. skip creating clientProfileMapping", pool.Name)
+		return nil
+	}
+
+	newClientProfileMapping, err := generateClientProfileMappingCR(clusterContext, clusterInfo, pool)
+	if err != nil {
+		return errors.Wrapf(err, "failed to generate clientProfileMapping for cephBlockPool %q", pool.Name)
+	}
+
+	err = createOrUpdateClientProfileMappingCR(clusterContext, clusterInfo, newClientProfileMapping)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create or update clientProfileMapping resource for cephBlockPool %q", pool.Name)
+	}
+
+	return nil
+}
+
+func createOrUpdateClientProfileMappingCR(clusterContext *clusterd.Context, clusterInfo *cephclient.ClusterInfo, clientProfileMapping *csiopv1a1.ClientProfileMapping) error {
+	existingClientProfileMapping := &csiopv1a1.ClientProfileMapping{}
+	request := types.NamespacedName{Name: clientProfileMapping.Name, Namespace: clusterInfo.Namespace}
```

Comment:
This should get created in the operator namespace https://github.com/rook/rook/pull/14551#issuecomment-2284146635

Reply:
Is this comment still  valid? I can see that `cephconnection and clientprofile CRs are not using operator namespace in master branch.

Reply:
its happening in this PR https://github.com/rook/rook/pull/14766